drivers/sensor/icm42670S/icm42670S_aml.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2023 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/sensor/icm42670S.h>
+
+#include "icm42670S.h"
+#include "invn_algo_aml.h"
+
+LOG_MODULE_REGISTER(ICM42670S_AML, CONFIG_SENSOR_LOG_LEVEL);
+
+#ifdef CONFIG_ICM42670S_AML
+static int32_t accel_mounting_matrix[9]= {0,   -1,    0,
+                                         -1,    0,    0,
+                                          0,    0,    1 };
+
+static int32_t gyro_mounting_matrix[9]= { 0,    1,    0,
+                                          1,    0,    0,
+                                          0,    0,   -1 };
+
+
+int icm42670S_aml_init(const struct device *dev, inv_imu_device_t *s, int8_t delta_gain_x, int8_t delta_gain_y)
+{
+	int rc = 0;
+	/* Contains algorithms configuration */
+	InvnAlgoAMLConfig config;
+	struct icm42670S_data *data = dev->data;
+	
+	/* FSR */
+	config.acc_fsr = 16;
```

Comment:
These values could be decided by the sample instead of hardcoded in the driver.
Not mandatory as setting max FSR for AML is a good practice.